docker/docker_image_dest.go

```diff
@@ -118,16 +119,16 @@ func (d *dockerImageDestination) PutBlob(digest string, stream io.Reader) error
 	uploadLocation.RawQuery = locationQuery.Encode()
 	res, err = d.c.makeRequestToResolvedURL("PUT", uploadLocation.String(), map[string][]string{"Content-Type": {"application/octet-stream"}}, stream)
 	if err != nil {
-		return err
+		return "", -1, err
 	}
 	defer res.Body.Close()
 	if res.StatusCode != http.StatusCreated {
 		logrus.Debugf("Error uploading layer, response %#v", *res)
-		return fmt.Errorf("Error uploading layer to %s, status %d", uploadLocation, res.StatusCode)
+		return "", -1, fmt.Errorf("Error uploading layer to %s, status %d", uploadLocation, res.StatusCode)
 	}
 
 	logrus.Debugf("Upload of layer %s complete", digest)
-	return nil
+	return "", -1, nil
```

Comment:
we need the digest before, I don't see how TeeReader could help, I'm wrapping something up with io.MultiWriter